pkg/instance/instance.go

```diff
@@ -59,3 +62,29 @@ func (i *Info) UpgradeRequired() bool {
 	// fully deconfigured before being configured by the current version.
 	return true
 }
+
+// GetIPv4Address returns an IPv4 address of the instance. An error will be thrown if the instance object has
+// an IPv6 address or a DNS address that does not resolve to an IPv4 network address.
+func (i *Info) GetIPv4Address() (string, error) {
+	if ip := net.ParseIP(i.Address); ip != nil {
+		// Address is either an IPv6 or IPv4 address
+		ipv4 := ip.To4()
+		if ipv4 == nil {
+			return "", errors.Errorf("IPv6 network address not supported: %s", ip.String())
+		}
+		return ipv4.String(), nil
+	}
+
+	// DNS address in this case
+	ips, err := net.LookupIP(i.Address)
+	if err != nil {
+		return "", errors.Wrapf(err, "lookup of address %s failed", i.Address)
+	}
+	// Get first IPv4 address returned
+	for _, returnedIP := range ips {
```

Comment:
This just occurred to me:
What if there are multiple IPv4 addresses and the first IPv4 address for whatever reason is not accessible but one of the others are? I guess this could also be a problem when the node is getting configured also and the question should have been asked where this was introduced.

Reply:
Addressed with new commit